src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalEntityScope.cs

```diff
@@ -18,6 +18,15 @@ internal interface IExternalEntityScope
 
         bool TryGetEntity<T>(DName currentEntityEntityName, out T externalEntity)
             where T : class, IExternalEntity;
+
+        /// <summary>
+        /// Checks if the given name is available or not.
+        /// <para>This is used in Tokenization to determine if namespace of the dotted name can be hidden or not.</para>
+        /// </summary>
+        /// <param name="name">Name.</param>
+        /// <param name="ignoreNamedFormulas">Flag indicating whether to ignore named formulas or not when checking for the availability of the given name.</param>
+        /// <returns>True if name is available and not used or false otherwise.</returns>
+        bool IsNameAvailable(string name, bool ignoreNamedFormulas = false);
```

Comment:
>bool IsNameAvailable(string name, bool ignoreNamedFormulas = false); [](http://example.com/codeflow?start=8&length=68)

Do we also need it in case of Dataverese?
[https://github.com/microsoft/Power-Fx-Dataverse/blob/main/src/PowerFx.Dataverse/CdsEntityMetadataProvider.cs#L32](https://github.com/microsoft/Power-Fx-Dataverse/blob/main/src/PowerFx.Dataverse/CdsEntityMetadataProvider.cs#L32)

If not, Can we have a new Interface implementing current and not break the Dataverse?

Reply:
That was my first thought but then i would need to change the definition of GlobalScope to implement the new interface instead of IExternalEntity in the backend of Canvas App. There are significant number of references to IExternalEntity in there and it would introduce a lot more compiler errors. I had to compromise somewhere so i thought of adding it to the IExternalEntity itself so whenever dataverse consumes the new power fx it could implement this one method. It won't break right away as dataverse would first need to consume new version after my changes are merged in. Anyways most of the methods of IExternalEntity in dataverse throw notsupportedexception so the implementation might be trivial.

Reply:
This is definitely the correct place to add this, fwiw. It's a method on the object that this was intended to be an interface for. We should look into getting DV off of this interface.

Reply:
Resolving this since putting this function directly in this interface is not a problem